src/highdicom/seg/sop.py

```diff
@@ -1913,48 +1903,305 @@ def _check_and_cast_pixel_array(
         return pixel_array, segments_overlap
 
     @staticmethod
-    def _omit_empty_frames(
-        pixel_array: np.ndarray,
-        plane_positions: Sequence[Optional[PlanePositionSequence]]
-    ) -> Tuple[np.ndarray, List[Optional[PlanePositionSequence]], List[int]]:
-        """Remove empty frames from the pixel array.
+    def _get_nonempty_plane_indices(
+        pixel_array: np.ndarray
+    ) -> Tuple[List[int], bool]:
+        """Get a list of all indices of original planes that are non-empty.
 
-        Empty frames (without any positive pixels) do not need to be included
-        in the segmentation image. This method removes the relevant frames
-        and updates the plane positions accordingly.
+        Empty planes (without any positive pixels in any of the segments) do
+        not need to be included in the segmentation image. This method finds a
+        list of indices of the input frames that are non-empty, and therefore
+        should be included in the segmentation image.
 
         Parameters
         ----------
         pixel_array: numpy.ndarray
             Segmentation pixel array
-        plane_positions: Sequence[Optional[highdicom.PlanePositionSequence]]
-            Plane positions for each of the frames
 
         Returns
         -------
+        included_plane_indices : List[int]
+            List giving for each plane position in the resulting segmentation
+            image the index of the corresponding frame in the original pixel
+            array.
+        is_empty: bool
+            Whether the entire image is empty. If so, empty frames should not
+            be omitted.
+
+        """
+        # This list tracks which source image each non-empty frame came from
+        source_image_indices = [
+            i for i, frm in enumerate(pixel_array)
+            if np.any(frm)
+        ]
+
+        if len(source_image_indices) == 0:
+            logger.warning(
+                'Encoding an empty segmentation with "omit_empty_frames" '
+                'set to True. Reverting to encoding all frames since omitting '
+                'all frames is not possible.'
+            )
+            return (list(range(pixel_array.shape[0])), True)
+
+        return (source_image_indices, False)
+
+    @staticmethod
+    def _get_segment_array(
```

Comment:
For consistency

```suggestion
    def _get_segment_pixel_array(
```

Reply:
resolved in [9092a00](https://github.com/ImagingDataCommons/highdicom/pull/227/commits/9092a000b3c1518b3c04c98b07d3b6682fb9a889)